Storage/Storage/SQL/JoinedHistoryVisitsTable.swift

```diff
@@ -155,7 +145,32 @@ class JoinedHistoryVisitsTable: Table {
             }
         }
 
-        // println("Query \(sql) \(args)")
+        // If the filter is a date type,
+        if dateQuery {
+            if let dateRange = options?.filter as? DateRange {
+                var addedWhere = false
+
+                // If we're querying with a date range, we want to do the filter after sites are grouped. We use
+                // a sub query for that.
```

Comment:
You probably don't!

Let me think about this some. The risk is that sqlite is going to do a LEFT JOIN over the entire DB, then filter by date in memory, each time you do a date-ranged query. That's crazy; to display the four history chunks on the homepage, it'd dump the entire DB into memory four times.